src/ehttpc.erl

```diff
@@ -426,77 +477,39 @@ gun_opts([{retry_timeout, _} | Opts], Acc) ->
     gun_opts(Opts, Acc);
 gun_opts([{connect_timeout, ConnectTimeout} | Opts], Acc) ->
     gun_opts(Opts, Acc#{connect_timeout => ConnectTimeout});
-gun_opts([{transport, Transport} | Opts], Acc) ->
-    gun_opts(Opts, Acc#{transport => Transport});
-gun_opts([{transport_opts, TransportOpts} | Opts], Acc) ->
-    gun_opts(Opts, Acc#{transport_opts => TransportOpts});
+gun_opts([{transport, Transport} | Opts0], Acc0) ->
+    Acc1 = Acc0#{transport => Transport},
+    case lists:keytake(transport_opts, 1, Opts0) of
+        {value, {_, TransportOpts}, Opts} when Transport == tcp ->
+            Acc = Acc1#{tcp_opts => TransportOpts};
+        {value, {_, TransportOpts}, Opts} when Transport == tls; Transport == ssl ->
+            Acc = Acc1#{tls_opts => TransportOpts};
```

Comment:
Q: are tcp options merged together for gun?

Reply:
Good question. AFAICS `gun` expects them to be separate. Added 2e92d4f0a2d7efa914d809d4758fe48e89f09618 to accommodate these expectations.